src/Loop.php

```diff
@@ -67,7 +80,8 @@ private static function createDriver()
         $driver = self::$factory->create();
 
         if (!$driver instanceof LoopDriver) {
-            throw new \LogicException("LoopDriverFactory didn't return a LoopDriver.");
+            $type = is_object($driver) ? "an instance of " . get_class($driver) : gettype($driver);
+            throw new \LogicException("Factory returned {$type}, but must return an instance of LoopDriver.");
```

Comment:
As there's no other factory, I think that's fine. But we could create another exception for it.